data/data.yaml

```
@@ -1270,3 +1270,162 @@
+    то происходит закладывание на стек, если прочитанный символ принадлежит алфавиту внутренних символов, то стек не меняется, 
+    если прочитанный символ принадлежит алфавиту возврата, то со стека происходит снятие верхнего элемента.
+    В PDA (Pushdown Automata), в отличие от VPA, со стеком можно совершать действия не зависимо от прочитанного символа обрабатываемой строки. Также
+    PDA также отличается функцией перехода: функция перехода в PDA позволяет переходить по пустому символу, но при этом изменять стек. 
+    Пример: язык чётных палиндромов не распознаётся VPA, однако он распознаётся PDA
+  author: Утебаева Милена
+
+- question: >
+    Формулировка леммы Ардена (Arden's rule).
+    Уравнения с регулярными выражениями
+  answer: > 
+    Лемма Ардена (Arden's rule): пусть X, Y, Z - регулярные выражения, X = YX | Z, где Z не зависит от X, пустое слово не принадлежит Y,
+    Тогда имеет место следующее равенство для языка L задаваемого регулярным выражением X: L(X) = Y^(*)Z, где оператор * является 
+    итерацией Клини (Kleene star).
+    Лемма Ардена позволяет решать системы уравнений в регулярных выражениях при помощи нетерминального исключения и подстановки, 
+    однако важно требованием является отсутствие цепных правил в грамматике
+  author: Утебаева Милена
+
+- question: >
+    Определение переключающегося автомата (Alternating Finite Automaton).
+    Автомат имеющий множество конъюнктивных состояний
+  answer: >
+    Переключающийся автомат A (Alternating Finite Automaton) задаётся шестью компонентами: A = (Σ, Q, Q_∀, F, q_0, δ), где Σ - алфавит,
+    Q - множество состояний автомата, Q_∀ - множество конъюнктивных состояний, F - множество конечных состояний,
+    q_0 - стартовое состояние, δ: (Σ ∪ e) × Q -> Q, то есть, область определения функции перехода - это алфавит, 
+    объединённый с пустым словом, а область значений функции перехода - это множество состояний.
+    Слово w допускается AFA (Alternating Finite Automaton) тогда и только тогда, когда существует путь 
+    q_0 -> q_1 -> ... -> q_n, причём слово из символов q_0q_1...q_n равно слову w, что если q_i принадлежит Q_∀ - множеству конъюнктивных 
+    состояний, и конфигурация имеет вид <q_i, v_i>, то есть, осталось прочитать суффикс v_i, то v_i распознаётся по 
+    всем переходам из q_i. Таким образом, здесь можно проследить аналогию с логическим «и»
+  author: Утебаева Милена
+
+- question: Определение конъюнктивной грамматики
+  answer: > 
+   Конъюнктивная грамматика G задаётся четырьмя компонентами: G = <Σ, N, S, Δ>, где Σ - алфавит, N - множество нетерминальных символов,
+   S - начальный символ, Δ - множество правил, назовём их A_i, таких, что они имеют вид: A_i -> φ_1 & φ_2 & ... & φ_n, 
+   где φ_j ∈ (Σ ∪ N)^(*), где каждый φ_j, называемый конъюнктом, состоит из терминальных и нетерминальных символов, 
+   символ & обозначает конъюнкцию или логическое «и».
+   Если строка S удовлетворяет всем условиям φ_1, φ_2, ..., φ_n, то S удовлетворяем правилу A_i 
```

```suggestion
   Если строка S удовлетворяет всем условиям φ_1, φ_2, ..., φ_n, то S удовлетворяет правилу A_i 
```

тут случайно не "т"?

---

я есть грут...
да, «т»

---

Лучше "удовлетворяет правилу разбора A_i", т.к. тут всё-таки нетерминалы, а иначе они будут путаться с названиями правил, хотя у одного и того же нетерминала может быть много правил разбора, и в этих обозначениях они все будут называться одинаково.